src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -120,9 +121,14 @@ private void setMsqQuestionDetails(FeedbackParticipantType generateOptionsFor, S
         this.msqChoices = new ArrayList<>();
         this.otherEnabled = false;
         this.generateOptionsFor = generateOptionsFor;
-        this.numOfMsqChoices = generateOptionList(courseId).size();
-        Assumption.assertTrue("Can only generate students, teams or instructors",
+        if (generateOptionsFor.equals(FeedbackParticipantType.STUDENTS_EXCLUDING_SELF)) {
```

Comment:
If I understand the change set correctly, you have added the implementation for `generateOptionList` for `STUDENTS_EXCLUDING_SELF`, which means this extra conditional should not be needed?

Reply:
In the original implementation, `generateOptionList(courseId).size()` would give an error because at this point of execution, the student is not passed into MsqQuestionDetails yet (the logic of execution for MCQ and MSQ seem to differ), so the method `generateOptionList` would result in NPE.

Reply:
Leaving a note here that this should be related to #8415.